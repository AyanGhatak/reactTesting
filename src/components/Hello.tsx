import * as React from 'react';
interface Hello {
  props: {
    name: string;
    enthusiasmLevel?: number;
    onIncrement?: () => void;
    onDecrement?: () => void;
  };
}

class Hello extends React.Component implements Hello {
  render() {
    const { name, enthusiasmLevel = 1, onIncrement, onDecrement  } = this.props;
    return (
      <div data-sumo-test="hello">
        <div className="greeeting">
          Hello {name + Array(enthusiasmLevel + 1).join('!')}
        </div>
        <button className="increment" onClick={onIncrement}> + </button>
        <button onClick={onDecrement}> - </button>
      </div>
    );
  }
}
export default Hello;
