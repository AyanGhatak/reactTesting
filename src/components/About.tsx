import * as React from 'react';
import Hello from './Hello';

interface About {
  state: {
    name: string;
    enthusiasmLevel?: number;
  };
}

interface Event {
  target: {
    value: string;
  };
}

class About extends React.Component implements About  {

  constructor(props: object) {
    super(props);
    this.state = {
      name: 'Ayan',
      enthusiasmLevel: 1,
    };
    this.onNameChange = this.onNameChange.bind(this);
    this.onIncrement = this.onIncrement.bind(this);
    this.onDecrement = this.onDecrement.bind(this);
  }

  onNameChange(e: Event) {
    this.setState({
      name: e.target.value,
    });
  }

  onIncrement() {
    this.setState((prevState: { enthusiasmLevel: number}) => ({ enthusiasmLevel: prevState.enthusiasmLevel + 1 }));
  }

  onDecrement() {
    this.setState((prevState: { enthusiasmLevel: number }) => ({
      enthusiasmLevel: Math.max(1, prevState.enthusiasmLevel - 1)
    }));
  }

  render() {
    const { name, enthusiasmLevel } = this.state;
    return (
      <div>
        <div><h2>About {name}</h2></div>
        <input value={name} type="text" onChange={this.onNameChange}/>
        <Hello
          name={name}
          onIncrement={this.onIncrement}
          onDecrement={this.onDecrement}
          enthusiasmLevel={enthusiasmLevel}
        />
      </div>
    );
  }
}

export default About;
