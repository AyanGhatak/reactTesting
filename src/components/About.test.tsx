import * as React from 'react';
import { shallow } from 'enzyme';
import About from './About';
import Hello from './Hello';

test('render a About', () => {
  const wrapper = shallow(<About />);
  expect(wrapper).toMatchSnapshot();
});

test('Should have 1 Hello Component', () => {
  const wrapper = shallow(<About />);
  expect(wrapper.find(Hello)).toHaveLength(1);
});
