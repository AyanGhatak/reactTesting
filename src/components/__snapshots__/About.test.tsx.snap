// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a About 1`] = `
<div>
  <div>
    <h2>
      About 
      Ayan
    </h2>
  </div>
  <input
    onChange={[Function]}
    type="text"
    value="Ayan"
  />
  <Hello
    enthusiasmLevel={1}
    name="Ayan"
    onDecrement={[Function]}
    onIncrement={[Function]}
  />
</div>
`;
